webbpsf/trending.py

```diff
@@ -313,11 +313,20 @@ def wfe_histogram_plot(
             elif ote_only is True:
                 opd_data = fits.getdata(full_file_path, ext=1)
                 mask = opd_data != 0
+                sensing_apername = fits.getheader(full_file_path, ext=0)['APERNAME']
 
-                # Get WSS Target Phase Map
-                was_targ_file = os.path.join(
+                # Get WSS Target Phase Map for the sensing aperture
+                # Note that the sensing maintenance program changed field point from NRC A3 to A1 around Dec 2024.
+                if sensing_apername == 'NRCA3_FP1':
+                    was_targ_file = os.path.join(
                     webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', 'wss_target_phase_fp1.fits'
                 )
+                elif sensing_apername == 'NRCA1_FP6':
+                    was_targ_file = os.path.join(
+                        webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', 'wss_target_phase_fp6.fits'
+                    )
```

Comment:
This if/elif statement is repeated five times, with no significant variation. This cries out to be refactored into a function, following the "Don't repeat yourself" principle. Here: 

```
def get_target_phase_map_filename(apername):
    """Get WSS Target Phase Map for the specified aperture
    Note that the sensing maintenance program changed field point from NRC A3 to A1 around Dec 2024.
    """
    if apername == 'NRCA3_FP1':
        fn = 'wss_target_phase_fp1.fits'
    elif sensing_apername == 'NRCA1_FP6':
        fn = 'wss_target_phase_fp6.fits'
    else:
        raise ValueError(f"Target phase map not available for aperture = {apername}")

    was_targ_file = os.path.join(
        webbpsf.utils.get_webbpsf_data_path(), 'NIRCam', 'OPD', fn)
    return was_targ_file
```

I suggest putting that in utils.py, then in each of the places in the code you can just call that function.

Reply:
Thanks, much appreciated your suggestion and I fully agree with you. I added the function to utils and change all the code accordingly.